### ISSUE TITLE
OS version major/minor truncated in UDFs

### ISSUE THREAD

**Issue by lucia-vargas-a** [mozilla/bigquery-etl#2917]
The UDF mozfun.norm.browser_version_info truncates to .1 every minor version that is .10, .100, .1000.

The problem originates in the usage of truncate_version, extract_version and browser_version_info which truncate to the float .1 in bigquery. Therefore it affects all UDFs that use them.

Here is one example of the problem:

![image](https://user-images.githubusercontent.com/30448600/165530765-6f9fec3b-9e74-4419-ba71-a459392b7853.png)

**Comment by ANich** [comment]
Good catch. I'm not sure how we can fix this for truncation while keeping the numeric type 🤔 3.10, 3.100 and 3.1000 always map to 3.1 numerically.

For extraction I think if we treat minor and patch versions as whole numbers we could get around this.. i.e. an update to `mozfun.norm.extract_version`

**Comment by lucia-vargas-a** [comment]
For the case of .10, .100, .1000 since numbers are not meant to have trailing zeros, how about providing both, the string and  the numeric values separated to allow calculations in Looker. Like this:

- As string:
version = 3.10.100
major = 3.10
minor = 3.10.100

- As numeric for calculations:
version = 3
major = 10
minor = 100

**Comment by sean-rose** [comment]
I think it'd make sense for `truncate_version` (and thus `browser_version_info`) to pad single-digit minor versions with a leading zero prior to converting to numeric so that minor versions compare/sort correctly with the two-digit minor versions we have.  For example, `browser_version_info("3.1").minor_version` would return `3.01`.

In my experience having minor versions more than two digits is rare, so I'd think that'd be sufficient, but if we have three-digit minor versions (or anticipate having any) we could pad minor versions further out to three digits.
